source/beamline/CLS/CLSBeamlineStatus.cpp

```diff
@@ -10,18 +13,20 @@ CLSBeamlineStatus::CLSBeamlineStatus(const QString &name, QObject *parent) :
 	monoMaskControlSet_ = new AMControlSet(this);
 
 	// Setup the basic value options.
-	addOption(On, "On", true);
-	addOption(Off, "Off", true);
+	addOption(On, "Beam is ON", true);
+	addOption(Off, "Beam is OFF", true);
+
+	connect(this, SIGNAL(connected(bool)), this, SLOT(updateBeamlineStatus()));
```

Comment:
@helfrij is there something in particular you don't like?  As long as there is no mechanism for signal-slot loops then this should be fine.